extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultConfigSource.java

```diff
@@ -0,0 +1,157 @@
+package io.quarkus.vault;
+
+import static io.quarkus.vault.LogConfidentialityLevel.MEDIUM;
+import static io.quarkus.vault.VaultAuthType.K8S;
+import static io.quarkus.vault.VaultServerRuntimeConfig.DEFAULT_CACHE_PERIOD;
+import static io.quarkus.vault.VaultServerRuntimeConfig.DEFAULT_K8S_TOKEN_PATH;
+import static io.quarkus.vault.VaultServerRuntimeConfig.DEFAULT_KV_SECRET_ENGINE_PATH;
+import static io.quarkus.vault.VaultServerRuntimeConfig.DEFAULT_RENEW_GRACE_PERIOD;
+import static io.quarkus.vault.VaultServerRuntimeConfig.KV_SECRET_ENGINE_VERSION_V1;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.configuration.DurationConverter;
+import io.quarkus.vault.service.dto.VaultToken;
+
+public class VaultConfigSource implements ConfigSource {
+
+    private static final Logger log = Logger.getLogger(VaultConfigSource.class);
+
+    private static final String QUARKUS_REST_CLIENT_URL = "io.quarkus.vault.VaultProxy/mp-rest/url";
```

Comment:
As mentioned earlier, it would be good to have a single property pointing to the vault endpoint shared between the simple proxies and `VaultService`

Reply:
it is the case. all you have to do is configure the io.quarkus.vault.url property, and mp property will be provisioned transparently for you.